### ISSUE TITLE
Add the len() function to JX

### ISSUE THREAD

**Issue by btovar** [cooperative-computing-lab/cctools#2214]
Given a JSON array A, len(A) should return the number of elements in the array.  For any other type, len should signal an error.  

(This should be done in dttools/src/jx_function.{h,c}.)

**Comment by dthain** [comment]
And just as important: update the documentation in jx-reference.md and jx-quick.md

**Comment by camicarballo** [comment]
How should I go about testing this to make sure it does what I am expecting it to do?

**Comment by btovar** [comment]
You can take a look at  dttools/test/TR_jx.sh.

Your input file would look something like:
```json
{
"count_1": len(["a", "b", "c"])
"count_2": len([])
}
count_1 == 3;
count_2 == 0;
```

with the expected file for comparison showing that both those expressions should be true.

**Comment by camicarballo** [comment]
It looks like the function jx_array_length is incorrect. For an array of any length it consistently returns a length of 1.

**Comment by btovar** [comment]
The iteration does there does seem suspect. Could you file an issue on it, and then change the iteration to use jx_iterate_array as shown in jx.h?   Thanks!

**Comment by btovar** [comment]
Fixed by #2216
